### PR TITLE
Emulated 3D textures

### PR DESCRIPTION
**Pull request by izaid**
Someone else should review this, especially the emulated 3D texture and its indexing schemes. It's worth trying for all kinds of crazy max_texture_sizes to make sure everything works. 

@rougier basically wrote the sample, so I'm ccing him here

### CODE REVIEW

**Comment by izaid** [comment]
Also ccing @almarklein, to make sure the volume rendering changes make sense.

File: examples/basics/gloo/animate_images_slice.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from vispy.util.transforms import ortho
 from vispy import gloo
```
Comment (almarklein) [inline examples/basics/gloo/animate_images_slice.py]: Add to the docstring that this uses emulated 3D?

File: vispy/gloo/texture.py [hunk 2]
```diff
@@ -185,6 +193,9 @@ def interpolation(self, value):
         self._glir.command('INTERPOLATION', self._id, *value)
     
     def resize(self, shape, format=None, internalformat=None):
+        return self._resize(shape, format, internalformat)
```
Comment (almarklein) [inline vispy/gloo/texture.py]: The public resize method should have a docstring.

**Comment by almarklein** [comment]
Other than minor comments. This LGTM. Awesome work. Thanks @izaid and @rougier!

File: examples/basics/scene/volume.py [hunk 3]
```diff
@@ -48,10 +48,13 @@
 # Set up a viewbox to display the image with interactive pan/zoom
 view = canvas.central_widget.add_view()
 
+# Set whether we are emulating a 3D texture
+emulated3d = True
+
 # Create the volume visuals, only one is visible
-volume1 = scene.visuals.Volume(vol1, parent=view.scene, threshold=0.5)
+volume1 = scene.visuals.Volume(vol1, parent=view.scene, threshold=0.5, emulated3d = emulated3d)
```
Comment (rossant) [inline examples/basics/scene/volume.py]: `emulated3d=emulated3d` (pep8)

**Pull request merged**